### PR TITLE
Add NikonFl7 makernotes group

### PR DESCRIPTION
**Pull request by postscript-dev**
`NikonFl7` is enabled for camera models where the `Exif.NikonFl7.version` tag is `0107` or `0108`. The tags are taken from ExifTool's [Nikon FlashInfo0107 tags](https://exiftool.org/TagNames/Nikon.html#FlashInfo0107).

This is part of resolving #1941.

`NikonFl7` differs from existing `NikonFl` groups as it uses `bigEndian` format. This is because `Exif.NikonFl7.ExternalFlashFirmware` is a short and was incorrectly output when using `invalidByteOrder` or `littleEndian`. The other groups use `invalidByteOrder` which may be wrong. Sadly all of Exiv2's test images for those groups don't use external flashes and the tag is set to zero.

Changes:
+ Add `NikonFl7` group
+ Add testing
+ Update documentation

Source: [Exiftool](https://exiftool.org/)

### CODE REVIEW

File: src/nikonmn_int.cpp [hunk 1]
```diff
@@ -2806,12 +2840,249 @@ fmountlens[] = {
...
+        std::ios::fmtflags f( os.flags() );
+        if (value.count() != 1 || value.typeId() != signedByte) {
+            os << "(" << value << ")";
+            os.flags(f);
+            return os;
+        }
+        std::ostringstream oss;
+        oss.copyfmt(os);
+        float temp = ( value.toFloat()/float(-6.0) );
+        temp *= float(1.00001);                 // Avoid round-off errors
```
[the remark this answers is not in the document]
Comment (kevinbackhouse) [reply src/nikonmn_int.cpp:2928]: @postscript-dev: I did a bit of experimenting and I don't think this code is working correctly. For example, if `temp == 1.5` then it prints `2/2`. Similarly when `temp == 0.666666` it prints `0/3`. I think the perl code in Exif.pm must have different rounding behavior than C++ does. Regardless, that perl code in Exif.pm looks numerically clumsy to me. My suggestion is to use a condition like this to check whether the number is suitable for printing as a fraction:

```c
std::abs(std::fmod(temp*3, 1)) < 0.001
```

I have attached the three files that I used for testing: [files.tar.gz](https://github.com/Exiv2/exiv2/files/7449010/files.tar.gz)

* `print_fraction.pm` is based on the code in [Exif.pm](https://github.com/exiftool/exiftool/blob/426df98011695c533f3b133fc9332c139aeaaa10/lib/Image/ExifTool/Exif.pm#L5043)
* `print_fraction1.cpp` is based on your code
* `print_fraction2.cpp` is my version that uses `std::abs` and `std::fmod`.

File: src/nikonmn_int.cpp [hunk 2]
```diff
@@ -2806,12 +2841,251 @@ fmountlens[] = {
...
+
+        if (temp == 0)
+            os << 0;
+        else if ( std::numeric_limits<float>::min() > temp || temp > std::numeric_limits<float>::max() )
```
Comment (kevinbackhouse) [inline src/nikonmn_int.cpp:2932]:
```suggestion
        else if (!std::isfinite(temp))
```

Comment (postscript-dev) [reply src/nikonmn_int.cpp]: I hadn't used this before but will in future. Thanks.

File: src/nikonmn_int.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (postscript-dev) [reply src/nikonmn_int.cpp:2928]: I haven't had much time recently and realised that I needed to look at this again. Thanks for helping.

File: src/nikonmn_int.cpp [hunk 3]
```diff
@@ -2806,12 +2841,251 @@ fmountlens[] = {
...
+        std::ostringstream oss;
+        oss.copyfmt(os);
+        float temp = ( value.toFloat()/float(-6.0) );
+        temp *= float(1.00001);                 // Avoid round-off errors
+
+        if (temp == 0)
+            os << 0;
+        else if (!std::isfinite(temp))
+            os << "(" << value << ")";
+        else if (std::abs(std::fmod(temp, 1)) < 0.001)
```
Comment (kevinbackhouse) [inline src/nikonmn_int.cpp:2934]:
```suggestion
        else if (std::abs(std::remainderf(temp, 1)) < 0.001)
```

File: src/nikonmn_int.cpp [hunk 4]
```diff
@@ -2806,12 +2841,251 @@ fmountlens[] = {
...
+            os << std::round(temp);
+        else if (std::abs(std::fmod(temp*2, 1)) < 0.001)
```
Comment (kevinbackhouse) [inline src/nikonmn_int.cpp:2936]:
```suggestion
        else if (std::abs(std::remainderf(temp*2, 1)) < 0.001)
```

File: src/nikonmn_int.cpp [hunk 5]
```diff
@@ -2806,12 +2841,251 @@ fmountlens[] = {
...
+            os << std::round(temp*2) << "/2";
+        else if (std::abs(std::fmod(temp*3, 1)) < 0.001)
```
Comment (kevinbackhouse) [inline src/nikonmn_int.cpp:2938]:
```suggestion
        else if (std::abs(std::remainderf(temp*3, 1)) < 0.001)
```

File: src/nikonmn_int.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (kevinbackhouse) [reply src/nikonmn_int.cpp:2928]: I made a mistake before. We need to use `std::abs(std::remainderf(...))`, not `std::abs(std::fmod(...))`. With that change, this multiplication is unnecessary. (The goal is that numbers like `0.4999` or `0.5001` are printed as a fraction.)
```suggestion
```

**Review by kevinbackhouse — CHANGES_REQUESTED** [review]

File: src/nikonmn_int.cpp (hunk 5 above)
Comment (postscript-dev) [reply src/nikonmn_int.cpp]: No problem, thanks for the help.

**Review by kevinbackhouse — APPROVED** [review]

**Pull request merged**